src/test/java/org/broadinstitute/consent/http/db/DatasetDAOTest.java

```diff
@@ -145,6 +147,131 @@ public void testFindNeedsApprovalDataSetByDataSetId() {
         assertTrue(datasets.get(0).getNeedsApproval());
     }
 
+    @Test
+    public void testGetNIHInstitutionalFile() {
+        Dataset dataset = createDataset();
+
+        createFileStorageObject(
+                dataset.getDataSetId().toString(),
+                FileCategory.ALTERNATIVE_DATA_SHARING_PLAN
+        );
+        FileStorageObject nihFile = createFileStorageObject(
+                dataset.getDataSetId().toString(),
+                FileCategory.NIH_INSTITUTIONAL_CERTIFICATION
+        );
+        createFileStorageObject();
+        createFileStorageObject(
+                dataset.getDataSetId().toString(),
+                FileCategory.DATA_USE_LETTER
+        );
+
+        Dataset found = datasetDAO.findDatasetById(dataset.getDataSetId());
+
+        assertEquals(nihFile, found.getNihInstitutionalCertificationFile());
+        assertEquals(nihFile.getBlobId(), found.getNihInstitutionalCertificationFile().getBlobId());
+    }
+
+    @Test
+    public void testGetNIHInstitutionalFile_AlwaysLatestUpdated() throws InterruptedException {
+        Dataset dataset = createDataset();
+
+
+        FileStorageObject nihFileCreatedFirstUpdatedSecond = createFileStorageObject(
+                dataset.getDataSetId().toString(),
+                FileCategory.NIH_INSTITUTIONAL_CERTIFICATION
+        );
+
+        Thread.sleep(10);
```

Comment:
I would prefer not to introduce sleeps to general tests. One way to get around this would be to create a local file storage insert method here in your test class that takes an Instant as a create date. That way, you could create instants of different times and use those instead of sleeping to force time passing.